flake.nix

```
@@ -0,0 +1,121 @@
+{
+  description = "`gitoxide` compiled in a nix shell, using `crane` and `flakebox`.";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-25.05";
+    flake-utils.url = "github:numtide/flake-utils";
+
+    fenix = {
+      url = "github:nix-community/fenix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    flakebox = {
+      url = "github:rustshop/flakebox";
+      inputs.nixpkgs.follows = "nixpkgs";
+      inputs.fenix.follows = "fenix";
+    };
+  };
+
+  outputs = { self, nixpkgs, flakebox, flake-utils, ... }:
+    flake-utils.lib.eachDefaultSystem (arch: let
+      pkgs = import nixpkgs { system = arch; };
+      flakeboxLib = flakebox.lib.mkLib pkgs { };
+
+      rustSrc = flakeboxLib.filterSubPaths {
+        root = builtins.path { name = "gitoxide"; path = ./.; };
+        paths = [ "Cargo.toml" "Cargo.lock" ".cargo"
```

As far as I can see, there is no `.cargo` file or directory in this repository. So filtering for it is confusing. Probably remove it?
```suggestion
        paths = [ "Cargo.toml" "Cargo.lock"
```